sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -138,29 +145,29 @@ class CostBasedOptimizer extends Optimizer with Logging {
         plan.childPlans.zip(childCosts).foreach {
           case (child, childCosts) =>
             val (childCpuCost, childGpuCost) = childCosts
-            val transitionCost = costModel.transitionToCpuCost(child)
+            val transitionCost = transitionToCpuCost(conf, child)
             val childGpuTotal = childGpuCost + transitionCost
             if (child.canThisBeReplaced && !consumesQueryStage(child)
                 && childGpuTotal > childCpuCost) {
               // force this child plan back onto CPU
               optimizations.append(ReplaceSection(
-                child.asInstanceOf[SparkPlanMeta[SparkPlan]], totalCpuCost, totalGpuCost))
+                child, totalCpuCost, totalGpuCost /*childCpuCost, childGpuTotal*/))
```

Comment:
Do we plan to uncomment these in the future?

Reply:
No. I pushed a change to remove those.